src/uu/wc/src/wc.rs

```diff
@@ -442,44 +426,38 @@ fn wc(inputs: Vec<Input>, settings: &Settings) -> Result<(), u32> {
         let word_count = match word_count_from_input(input, settings) {
             CountResult::Success(word_count) => word_count,
             CountResult::Interrupted(word_count, error) => {
-                show_error(input, error);
-                failure = true;
+                show!(uio_error!(error, "{}", input.path_display()));
```

Comment:
The `uio_error!()` macro calls `UIoError::new()` with `error.kind()` as its first argument: https://github.com/uutils/coreutils/blob/0f2d8c597d455ea9467db3e213a6794c97a7c306/src/uucore/src/lib/mods/error.rs#L548-L551

The `UIoError::new()` function calls `kind.into()` to re-create a new instance of `std::io::Error`: https://github.com/uutils/coreutils/blob/0f2d8c597d455ea9467db3e213a6794c97a7c306/src/uucore/src/lib/mods/error.rs#L383

When it comes time to `Display` the `UIoError` instance, it checks if the inner `std::io::Error` instance has `raw_os_error()` set to `Some`: https://github.com/uutils/coreutils/blob/0f2d8c597d455ea9467db3e213a6794c97a7c306/src/uucore/src/lib/mods/error.rs#L397

However, at this point the call to `raw_os_error()` now returns `None` because of the conversion from `std::io::Error` to `ErrorKind` back to `std::io::Error`.

I could work around this by just not using the `UIoError` type here at all and displaying the message manually here in the `wc.rs` file, but it would be great if someone sees a straightforward change that could be made to `UIoError` that would allow this to work?

Reply:
Yeah this is also an issue in the most recent `ls` PR. I think that behaviour should be reverted because it is quite unintuitive. We just have to be careful to also think of a workaround for the case that behaviour was introduced for.

Reply:
For now, I've just added some code to use `USimpleError` and workaround this deficiency.